src/axom/inlet/tests/inlet_Inlet.cpp

```diff
@@ -1058,8 +1058,30 @@ TEST(inlet_Inlet_verify, verifyValidStringValues) {
   field = inlet1->addString("NewTable/field5");
   field->validValues(strs)->defaultValue("zyx");
   EXPECT_FALSE(inlet1->verify());
+}
+
+TEST(inlet_verify, verifyLambda) {
+  std::string testString = "field1 = true; field2 = 'abc'; NewTable = { field3 = 'xyz'; field4 = 'yes' }";
+  DataStore ds;
+  auto inlet = createBasicInlet(&ds, testString);
+
+  auto field1 = inlet->addBool("field1");
+  auto field2 = inlet->addString("field2");
+  auto field3 = inlet->addString("NewTable/field3");
```

Comment:
Instead of having the user implement the `getView` + cast to appropriate type, possibly they could optionally specify a verifier when they add a field, e.g.
```cpp
auto field2 = inlet->addString("field2", [](const std:string& field) { return str.size() >= 1 && str[0] == 'a'; });
```
with signature `addString(name, std::function<bool(string)> verifier, desc)`

...and then wrap up the boilerplate getView + cast when you register the lambda, since you'd know both the name of the view and the type you'd need to cast it to.

Reply:
an alternative is to follow the ongoing convention of chaining functions like so:

`auto field2 = inlet->addString("field2")->addVerifier([](const std:string& str) { return str.size() >= 1 && str[0] == 'a'; });`

That's assuming the only thing users would want to verify is the value.  Which might be true. 

I would like to avoid users having to do sidre calls but maybe that's wishful thinking.

@kennyweiss  thoughts?

Reply:
I agree with @white238 -- 
* The `Field` class should have an `addVerifier()` function that takes a `std::function` (which could also be a `lambda`).
* The `field` has sufficient context to know the path to the associated group, so that does not need to be passed in
* The sidre abstraction should definitely not be exposed to the user -- that is an implementation detail that they shouldn't have to be concerned with.

Does the verifier need any additional parameters, in general?

Reply:
The way `cli11` does this makes sense to me: https://github.com/CLIUtils/CLI11#validators

They have a function called `check()` that takes a `Validator` object, and there are several built-in validators.

For `Inlet`, it would make sense to start with a lambda without parameters. If more structure is needed, perhaps this can be extended to an abstract base class that can be used to validate the fields.

Bonus points to consider for the future: It would be **really** cool if you could define the validators in a scripting language like lua.  (I'm not entirely sure how that would work though...)

Reply:
I agree that the Sidre details should ideally not be revealed to users. If lambda verifiers would only be needed for checking the values of Fields and no other properties, I can go ahead and add addVerifier(). However, if we would like to allow users to verify multiple Field properties, the above approaches wouldn't allow that flexibility. Is providing lambda verifiers for only Field values sufficient?

Reply:
Fair enough.  

In these cases, it almost always makes sense to brainstorm some expected use cases and program for those, rather than something generic that will work in all possible cases.

I'd imagine that most validators would be on a single field. Perhaps the more general ones can be validators on an entire `Table`?

Reply:
I think there is probably multiple levels of verifiers we want to add...

* Specific Field
* Specific Table
* a global one that works on all Fields
* same but on Tables
* Then one that exposes the sidre root group and the users is expected to know what they are doing.